Prefer HTTPS if available for package sources

dcousens opened the pull request:
This pull request changes a number of `PKG_SOURCE_URL`s that were using `http:`, to use `https:` if they were available.

I understand that this shouldn't impact the integrity of the downloaded package, which has a `PKG_HASH` for that purpose.
This is none-the-less a beneficial change as HTTPS becomes a modern baseline.

dcousens commented on package/utils/bsdiff/Makefile:
```diff
@@ -27,15 +27,15 @@ define Package/bsdiff
   DEPENDS:=+libbz2
   TITLE:=Binary diff tool
-  URL:=http://www.daemonology.net/bsdiff/
+  URL:=https://www.daemonology.net/bsdiff/
```
This isn't a package source, but consistency

dcousens commented:
Resolved commit formalities

Ansuel commented:
@dcousens thanks for the pr. Commit description are mandatory so pls add it. Extra bonus point if you put in the description the update packages :D

Ansuel commented:
Also hope it's not a problem but currently we don't accept github no-replay email :(

robekl commented:
This will break transparent http caching

Ansuel commented:
@robekl can you elaborate? majority of packet have https url also we are starting to check certificate.

dcousens commented:
@Ansuel updated description and signed off with a different email (that poor inbox)

Commit:
build: prefer HTTPS if available (for packages)

Changes PKG_SOURCE_URL's for arptables, bsdiff, dnsmasq,
fortify-headers, ipset, ipset-dns, libaudit, libpcap, libressl,
lua, lua5.3, tcpdump and valgrind, to HTTPS

Signed-off-by: Daniel Cousens <github@dcousens.com>

dcousens commented:
Hopefully third time :pray:

Had to disable a few GitHub options to make this work

Ansuel commented:
@dcousens what options you had to disable o.O

dcousens commented:
Sorry, 1 option, the other option wasn't relevant in the end.

>  Block command line pushes that expose my email

Didn't have to use that when using `+openwrt`, but to fix the authorship error I needed to use an associated email to the account.
Ah well, that inbox is practically blocked nowadays anyway, non-stop phishing.

diizzyy commented:
Unless sites actually redirects to (forces) https this is just for the sake of it and doesn't provide any benefit, it's been discussed many times before.

Ansuel commented:
@diizzyy we now check the certificate as an additional security measure so it would make sense... even just for consistency. Are you against it or just neutral? I would honestly merge this and be done with it.

dcousens commented:
> Unless sites actually redirects to (forces) https 

This changes the download to use `https`, there needn't be any redirection.
You might find some `http` routes are redirect to `https`, but why rely on that if we _know_ https is available?

diizzyy commented:
@Ansuel 
By that you mean that it's validated and not ignored? That might bite you since OpenWrt tends to "support" very old hosts for a very long time.

Ansuel commented:
@diizzyy with this https://github.com/openwrt/openwrt/pull/4948 we don't ignore https cert anymore. The option can be disabled from menuconfig but this is enabled by default.

diizzyy commented:
Yeah, jow- also raised concerns about that but oh well...

Ansuel commented:
@diizzyy true but if you think about it on a different view, host with expired cert should be replaced and if possible fixed. Also it will permit us to discover if our own cdn have problems with some certificate.
It can cause problems but ignoring the cert looks to be more a workaround than an intended thing.
(didn't want to start a discussion just giving you some reason why i merged that)

diizzyy commented:
As much as we can argue about that OpenWrt have a history for keeping support for "legacy" platforms for a very long time so I'm just saying that it may clash with how releases are handled and supported.

dcousens commented:
@diizzyy if we added fallbacks to `http:`, would that alleviate your concerns?
Without this change, I couldn't actually download these resources as the network environment I am in blocks HTTP outbound, only allowing HTTPS.

hauke commented:
I do not understand the problem with using https by default?
If the original server is not available any more or the certificate is broken the build system will use one of our own mirrors automatically.
If you have an old build host with an outdated trust store you can run into problems, but distributions normally update it for systems still under support.

diizzyy commented:
@dcousens 
That doesn't help anything at all just adding more workarounds.

@hauke
It would make more sense in that regard to add an option to opt-in regarding certificates.

That being said, I don't really care but I do see how this can cause breakage.
